src/components/button.js

```diff
@@ -0,0 +1,47 @@
+var React = require('react/addons');
+
+var Button = React.createClass({
+	propTypes: {
+		label: React.PropTypes.string.isRequired,
+		type: React.PropTypes.oneOf([
+			'default',
+			'primary',
+			'positive',
+			'negative',
+			'link'
+		]),
+		outlined: React.PropTypes.bool,
+		block: React.PropTypes.bool,
+		icon: React.PropTypes.element,
+		badge: React.PropTypes.element
```

Comment:
Both `Icon` and `Badge` will be actual components. We can use `this.props.children` to offer support for both without having to add them as `props`.